x/nexus/keeper/msg_server.go

```diff
@@ -97,3 +102,152 @@ func (s msgServer) DeregisterChainMaintainer(c context.Context, req *types.Dereg
 
 	return &types.DeregisterChainMaintainerResponse{}, nil
 }
+
+func (s msgServer) ActivateChain(c context.Context, req *types.ActivateChainRequest) (*types.ActivateChainResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	for _, chainStr := range req.Chains {
+		chain, ok := s.GetChain(ctx, chainStr)
+		if !ok {
+			return nil, fmt.Errorf("%s is not a registered chain", chainStr)
+		}
+
+		if s.IsChainActivated(ctx, chain) {
+			continue
+		}
+
+		sumConsensusPower := sdk.ZeroInt()
+		maintainers := s.GetChainMaintainers(ctx, chain)
+
+		for _, maintainer := range maintainers {
+			validator := s.staking.Validator(ctx, maintainer)
+			if validator == nil {
+				continue
+			}
+
+			if !validator.IsBonded() || validator.IsJailed() {
+				continue
+			}
+
+			sumConsensusPower = sumConsensusPower.AddRaw(validator.GetConsensusPower(s.staking.PowerReduction(ctx)))
+		}
+
+		if utils.NewThreshold(sumConsensusPower.Int64(), s.staking.GetLastTotalPower(ctx).Int64()).GTE(s.GetParams(ctx).ChainActivationThreshold) {
+			s.Nexus.ActivateChain(ctx, chain)
+
+			s.Logger(ctx).Info(fmt.Sprintf("chain %s activated", chain.Name))
+			ctx.EventManager().EmitEvent(
+				sdk.NewEvent(
+					types.EventTypeChain,
+					sdk.NewAttribute(sdk.AttributeKeyModule, types.ModuleName),
+					sdk.NewAttribute(sdk.AttributeKeyAction, types.AttributeValueActivated),
+					sdk.NewAttribute(types.AttributeKeyChain, chain.Name),
+				),
+			)
+		}
+	}
+
+	return &types.ActivateChainResponse{}, nil
+}
+
+// FreezeChain handles freeze a chain in case of emergencies
+func (s msgServer) FreezeChain(c context.Context, req *types.FreezeChainRequest) (*types.FreezeChainResponse, error) {
```

Comment:
if internally we refer to chain as activated/deactivated, then the commands/funcs calls should also be named "activate/deactivate".